modules/Header/ui/CategoriesNav/CategoriesNavDesktop.tsx

```diff
@@ -127,23 +132,31 @@ export function CategoriesNavDesktop({
                                 <div className={styles.backdrop} />
                             </div>
                         </Transition>
-                        {open && <BlockPageScroll />}
+                        {open && <BlockPageScroll scrollBarWidth={scrollBarWidthRef.current} />}
                     </>
                 )}
             </Popover>
         </li>
     );
 }
 
-function BlockPageScroll() {
+function BlockPageScroll({ scrollBarWidth }: BlockPageScroll.Props) {
     useEffect(() => {
         document.body.classList.add(styles.preventScroll);
-
-        return () => document.body.classList.remove(styles.preventScroll);
-    }, []);
+        document.body.style.marginRight = `${scrollBarWidth}px`;
+        return () => {
+            document.body.classList.remove(styles.preventScroll);
+            document.body.style.marginRight = `0`;
+        };
+    });
```

Comment:
Should `scrollBarWidth` be declared as the effect dependency?

Reply:
Yes it should help in getting rid of any unnecessary rerenders.